hal/src/samd51/usb/bus.rs

```diff
@@ -326,10 +327,18 @@ impl<'a> Bank<'a, InBank> {
             desc.set_byte_count(0);
         }
 
-        self.epstatusclr(idx)
-            .write(|w| w.stallrq1().set_bit().dtglin().set_bit().bk1rdy().set_bit());
-        self.epintenset(idx).write(|w| w.trcpt1().set_bit());
-        self.clear_transfer_complete();
+        if idx == 0 {
```

Comment:
Is this scope necessary at all? From manual


        //The USB bus reset is initiated by a connected host and MANAGED BY HARDWARE.
        // During USB reset the following registers are cleared:
        // Device Endpoint Configuration (EPCFG) register - except for Endpoint 0
        // Device Frame Number (FNUM) register
        // Device Address (DADD) register
        // Device Endpoint Interrupt Enable Clear/Set (EPINTENCLR/SET) register
        // Device Endpoint Interrupt Flag (EPINTFLAG) register
        // Transmit Stall 0 bit in the Endpoint Status register (EPSTATUS.STALLRQ0)
        // Transmit Stall 1 bit in the Endpoint Status register (EPSTATUS.STALLRQ1)
        // Endpoint Interrupt Summary (EPINTSMRY) register

Reply:
I think so, I put that conditional in so that we reset those bits on endpoint zero only, which is the endpoint which does not get cleared.

Reply:
Wait did I read that wrong? The EPSTATUS register is reset for all endpoints?